Index: Scripts/app/ViewModels/disasterSelectionListAdminArea.js
===================================================================
--- Scripts/app/ViewModels/disasterSelectionListAdminArea.js	(revision 10643)
+++ Scripts/app/ViewModels/disasterSelectionListAdminArea.js	(working copy)
@@ -6,7 +6,12 @@
     this.financialFormList = ko.observableArray();
     
     this.navigateToForm = function(item){
-      SingletonAppViewModel.getInstance().loadViewModule('FinancialForm', item);
+      if (item.Status) {
+        SingletonAppViewModel.getInstance().loadViewModule('FinancialFormEdit', item.Status.Id);
+      }else {
+        SingletonAppViewModel.getInstance().loadViewModule('FinancialForm', item);
+      }
+      
     };
 
     this.navigateToRequests = function(){
Index: Scripts/app/ViewModels/FinancialForm/request/financialForm.js
===================================================================
--- Scripts/app/ViewModels/FinancialForm/request/financialForm.js	(revision 10643)
+++ Scripts/app/ViewModels/FinancialForm/request/financialForm.js	(working copy)
@@ -11,6 +11,9 @@
     var financialCategoryObject = new FinancialCategory();
     var dependantsCategoryObject = new DependantsCategory();
     this.disasterItem = !$.isEmptyObject(data.disasterItem)? data.disasterItem : {Name:'', Id:0};
+    if (data.editId) {
+      alert('Editing');
+    }
     this.financialFormId = ko.observable(0);
     this.isLoading = ko.observable(false);
 
@@ -138,4 +141,26 @@
     };
   };
   return FinancialForm;
-});
\ No newline at end of file
+});
+/*
+var vale = 'no';
+$('input:radio[value="' + vale + '"][name="disabilities"]').iCheck('check');
+
+***********
+this.itemCategoryList()
+push Item
+Object {Type: 1, Name: "Cat5", Amount: 500}
+this automatically will update totalEstimatedLoss computed
+************
+name: 'FinancialForm',
+view: 'FinancialForm',
+routeAssociated: '',
+-----
+name: 'FinancialFormEdit',
+view: 'FinancialForm',
+routeAssociated: 'request/:id',
+module: 'ViewModels/financialForm/request/financialForm',
+onModuleLoaded: _.bind(function (resolveData, Module) {
+  resolveData(new Module({editId: this.params}));
+},this)
+*/
\ No newline at end of file
