Index: Scripts/app/applicationManager/RegionHandler.js
===================================================================
--- Scripts/app/applicationManager/RegionHandler.js	(revision 10691)
+++ Scripts/app/applicationManager/RegionHandler.js	(working copy)
@@ -13,7 +13,7 @@
           name: moduleData.name,
           key:key,
           route: moduleData.routeAssociated,
-          routePattern: this.routeToRegExp(moduleData.routeAssociated)
+          routePattern: this.routeToRegExp(moduleData.routeAssociated || '')
         };
       }, this));
     };
Index: Scripts/app/ViewModels/FinancialForm/request/lossCategory.js
===================================================================
--- Scripts/app/ViewModels/FinancialForm/request/lossCategory.js	(revision 10691)
+++ Scripts/app/ViewModels/FinancialForm/request/lossCategory.js	(working copy)
@@ -140,7 +140,7 @@
     }, this);
     
     //DRY
-    this.itemFunction = _.bind(function(inputCheckbox, item){
+    /*this.itemFunction = _.bind(function(inputCheckbox, item){
       if (inputCheckbox.checked) {
         $('#' + 'carItem' + item.Id).val('').prop('disabled', true);
         this.carItemChanged();
@@ -157,7 +157,24 @@
       }else{
         $('#' + 'houseItem' + item.Id).prop('disabled', false);
       }
-    }, this);
+    }, this);*/
+
+    this.ifun = function(inputPrefix, propCalc){
+      return _.bind(function(input, item){
+        if (input.checked) {
+          $('#' + inputPrefix + item.Id).val('').prop('disabled', true);
+          this[propCalc]();
+        }else{
+          $('#' + inputPrefix + item.Id).prop('disabled', false);
+        }
+      }, this);
+    };
+
+    this.itemFunction = this.ifun('carItem', 'carItemChanged');
+
+    this.itemCheckboxFunction = this.ifun('houseItem', 'houseItemChanged');
+
+    
   };
   return LossCategory;
 });
\ No newline at end of file
